docs/template.rst

```diff
@@ -0,0 +1,93 @@
+.. _templates:
+
+Templates
+=========
+
+This document describes how to use templates with Django Bread.
+
+Template Context
+----------------
+
+In addition to what Django Vanilla Views provides, Django Bread provides the following to all of
+your templates::
+
+        Variable             Description
+        --------             -----------
+        bread                Bread object
+        verbose_name         Model's verbose_name field
+        verbose_name_plural  Model's verbose_name_plural field
+        base_template        The default base_template that should be extended.
+        may_{action}         (i.e. may_browse, may_read, etc.) Boolean describing whether user has specified permission
+
+Certain action views provide some additional variables. The Browse view has ``columns`` and
+``filter``. The Read view has ``form``.
+
+Override ``get_context_data`` to change the variables provided to your template:
+
+.. code-block:: python
+
+   class MyBrowseView(BrowseView):
+       def get_context_data(self, **kwargs):
+           data = super(BreadViewMixin, self).get_context_data(**kwargs)
+           data['my_special_var'] = 42
+           return data
+
+
+Template Resolution
+-------------------
+
+There are various ways to configure how and in what order Django Bread searches for the proper
+template for your view.
+
+By default, each Django Bread view will search first in an app-specific location (described below),
+and then fall back to the templates provided by the Django Bread package.
+
+App-specific location
+^^^^^^^^^^^^^^^^^^^^^
+
+The app-specific location is described by this pattern: ``{app_name}/{model_name}{activity}.html``.
```

Comment:
Could we document this pattern using the same syntax and variables as the user would use to specify their own custom template name patter?